pkg/domain/infra/abi/images.go

```diff
@@ -727,24 +724,25 @@ func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entitie
 			sigStoreDir = options.Directory
 		}
 		if sigStoreDir == "" {
-			if rootless.IsRootless() {
-				sigStoreDir = filepath.Join(filepath.Dir(ir.Libpod.StorageConfig().GraphRoot), "sigstore")
-			} else {
-				var sigStoreURI string
-				registryInfo := trust.HaveMatchRegistry(rawSource.Reference().DockerReference().String(), registryConfigs)
-				if registryInfo != nil {
-					if sigStoreURI = registryInfo.SigStoreStaging; sigStoreURI == "" {
-						sigStoreURI = registryInfo.SigStore
-					}
-				}
-				if sigStoreURI == "" {
-					return nil, errors.Errorf("no signature storage configuration found for %s", rawSource.Reference().DockerReference().String())
-
+			var sigStoreURI string
+			registryInfo := trust.HaveMatchRegistry(rawSource.Reference().DockerReference().String(), registryConfigs)
+			if registryInfo != nil {
+				if sigStoreURI = registryInfo.SigStoreStaging; sigStoreURI == "" {
+					sigStoreURI = registryInfo.SigStore
 				}
-				sigStoreDir, err = localPathFromURI(sigStoreURI)
-				if err != nil {
-					return nil, errors.Wrapf(err, "invalid signature storage %s", sigStoreURI)
+			}
+			if sigStoreURI == "" {
+				// set the default sigstore for rootless or root if it is not configured
+				if rootless.IsRootless() {
+					sigStoreURI = trust.RootlessSignatureStoreURL()
+				} else {
+					sigStoreURI = trust.DefaultSignatureStoreURL()
```

Comment:
Dealing with URLs here seems to be an unnecessary complication; move the `localPathFromURI` call to the `registryInfo != nil` case, and this can just work with the `sigStoreDir` path.